examples/03_connectivity/plot_rest_clustering.py

```diff
@@ -30,130 +51,148 @@
-# Second, we illustrate the effect that the clustering has on the
-# signal. We show the original data, and the approximation provided by
-# the clustering by averaging the signal on each parcel.
-#
-# As you can see below, this approximation is very good, although there
+plotting.plot_epi(index_img(fmri_compressed[0], 0),
+                  cut_coords=cut_coords,
+                  title='Ward compressed representation (2000 parcels)',
+                  vmin=vmin, vmax=vmax, display_mode='xz')
+# As you can see below, this approximation is almost good, although there
 # are only 2000 parcels, instead of the original 60000 voxels
 
-# Display the original data
-plot_epi(nifti_masker.inverse_transform(fmri_masked[0]),
-         cut_coords=cut_coords,
-         title='Original (%i voxels)' % fmri_masked.shape[1],
-         vmax=fmri_masked.max(), vmin=fmri_masked.min(),
-         display_mode='xz')
+#########################################################################
+# Brain parcellations with KMeans Clustering
+# ------------------------------------------
+#
+# We use the same approach as demonstrated with building parcellations using
+# Ward clustering.
+
+# class/functions can be used here as they are already imported above.
+
+# This object uses method='kmeans' for KMeans clustering with 6mm smoothing
+kmeans = Parcellations(method='kmeans', n_parcels=100,
+                       standardize=False, smoothing_fwhm=6.,
+                       memory='nilearn_cache', memory_level=1,
+                       verbose=1)
+# Call fit on functional dataset: single subject (less samples)
+kmeans.fit(dataset.func)
+print("KMeans 2000 clusters: %.2fs" % (time.time() - start))
+
+# NOTE: Good parcellations can be build using KMeans with more subjects,
+# for instance more than 5 subjects. Follow pointed references for good
+# arguments on using n_parcels and the amount of smoothing, etc.
+###########################################################################
+# Visualize: Brain parcellations (KMeans)
+# ---------------------------------------
+#
+# Grab parcellations of brain image stored in attribute `labels_img_`
+kmeans_labels_img = kmeans.labels_img_
 
-# A reduced data can be create by taking the parcel-level average:
-# Note that, as many objects in the scikit-learn, the ward object exposes
-# a transform method that modifies input features. Here it reduces their
-# dimension
-fmri_reduced = ward.transform(fmri_masked)
+plotting.plot_roi(kmeans_labels_img, mean_func_img,
+                  title="KMeans parcellation",
+                  display_mode='xz')
 
-# Display the corresponding data compressed using the parcellation
-fmri_compressed = ward.inverse_transform(fmri_reduced)
-compressed_img = nifti_masker.inverse_transform(fmri_compressed[0])
+# kmeans_labels_img is a Nifti1Image object, it can be saved to file with
+# the following code:
+kmeans_labels_img.to_filename('kmeans_parcellation.nii')
```

Comment:
I try to always use a "nii.gz" in the examples, to teach users to use it, as it saves space on the disk.